app/lib/server/methods/sendMessage.js

```diff
@@ -63,8 +63,8 @@ export function executeSendMessage(uid, message) {
 	// do not allow nested threads
 	if (message.tmid) {
 		const parentMessage = Messages.findOneById(message.tmid);
-		message.tmid = parentMessage.tmid || message.tmid;
-		rid = parentMessage.rid;
+		message.tmid = (parentMessage && parentMessage.tmid) || message.tmid;
```

Comment:
why?

Reply:
The problem is that tasks are not messages under the hood. However, tasks can become header threads. And so the parentMessage became undefined. 

Maybe i should find a way to find a parentTask as well depending on whether we add a message under a task.